### DISCUSSION TITLE
Using Data Processor integer value (decimal not HEX)

### DISCUSSION THREAD

**Discussion by Uzver123** [WerWolv/ImHex#1551] [Get Help]
I want to display decimal value how i do it?

![1](https://github.com/WerWolv/ImHex/assets/143356096/ee22d68d-0a58-4fb3-b5ae-78b1e6a2a9f6)

**Comment by paxcut** [comment]
you can display the values in the hex editor as 32 bit decimals. to do so click on the up chevron (where it says region in your post above) and on the right you'll see a menu to choose data types to seethe data. There is no way currently to have the data processor node show decimal numbers that I am aware of.

![image](https://github.com/WerWolv/ImHex/assets/53811119/ea469eeb-144a-4dea-8d2b-beced542b48c)

**Comment by Uzver123** [comment]
Sad i want to replace only some values to decimal representation not entire file.

**Comment by Uzver123** [comment]
There is no way to replace only 4 bytes of hex with decimal representation?

![image](https://github.com/WerWolv/ImHex/assets/143356096/b51f7207-f85f-4201-bad1-1e97f8658d8d)

**Reply by Uzver123** [reply]
Is there way to selectively replace values inside Hex editor tab using `patterns`.

Once pattern is created it shows value in `pattern data` tab, is there way to replace value in the Hex editor green with value from `pattern`?

![image](https://github.com/WerWolv/ImHex/assets/143356096/a126cbd3-6147-4d1d-8e0f-e43cc339c424)

**Reply by paxcut** [reply]
you cannot display one value in decimal and the rest in hexadecimal. Is that feature present in other hex editors? If something like that existed, how does one know what base the number displayed has?

**Comment by Uzver123** [comment]
> If something like that existed, how does one know what base the number displayed has?

Can use specific background color to understand that this is replacement.


> Is that feature present in other hex editors?

Not that i know off, that's the beauty of it can render some blocks either 1 byte size or select size as decimal to improve readability of data once you already structured it. 

Should not be a problem to program this in since currently already possible to replaces text in Hex editor using `data processor`.

Here i copied first 16 bytes to position 96 in hex editor

![image](https://github.com/WerWolv/ImHex/assets/143356096/fdf687b4-7ab2-4d52-b086-12051d4515aa)

Most of the time i want to show decimal so its not a big deal if it will replace hex decimal values `07 E8` to `20 24`
